system/jlib/jstats.cpp

```diff
@@ -1988,9 +2000,58 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             cur.visit(visitor);
     }
 
+    virtual bool refreshAggregates(std::vector<StatisticKind> & aggregateKinds, std::vector<unsigned __int64> & totals) override
```

Comment:
Would a CRuntimeStatisticCollection (instead of aggregateKinds, totals) simplify this code?  The dual arrays seem to implement the same idea.

Reply:
CRuntimeStatisticCollection cannot store the variants.  When the stats are deserialized, we'd have to ignore all the variants.  We could ignore all the variants.  Losing the variants would mean that we could not have a single in-memory object for the stats as we'd need one for aggregation (GlobalStatisticCollection ) and another for serialization(CStatisticGatherer).  We would also not be able to extend the functionality to use GlobalStatisticCollection to cache runtime stats and allow periodic updates to dali.

Reply:
Also, not that a new CRuntimeStatisticCollection will need to be created for each scope and each one will have a StatisticMapping. With the current implementation, a single vector with the StatisticKinds is used as it traverses the entire StatisticsCollection.  I think using CRuntimeStatisticCollection will be slower and use more memory.